rego/rego.go

```diff
@@ -1401,64 +1465,32 @@ func (r *Rego) Compile(ctx context.Context, opts ...CompileOption) (*CompileResu
 		queries = []ast.Body{r.compiledQueries[compileQueryType].query}
 	}
 
-	return r.compileWasm(modules, queries, compileQueryType)
-}
-
-func (r *Rego) compileWasm(modules []*ast.Module, queries []ast.Body, qType queryType) (*CompileResult, error) {
-	decls := make(map[string]*ast.Builtin, len(r.builtinDecls)+len(ast.BuiltinMap))
-
-	for k, v := range ast.BuiltinMap {
-		decls[k] = v
-	}
-
-	for k, v := range r.builtinDecls {
-		decls[k] = v
+	if tgt := r.targetPlugin(r.target); tgt != nil {
+		return nil, fmt.Errorf("unsupported for rego target plugins")
```

Comment:
Can you please help me understand why this check is done.

Reply:
It's meant to prevent compilation when using as plugin. We could support it later...